src/librustc_mir/interpret/eval_context.rs

```diff
@@ -881,47 +878,77 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         self.raw_const_to_mplace(val)
     }
 
-    pub fn dump_place(&self, place: Place<M::PointerTag>) {
-        // Debug output
-        if !log_enabled!(::log::Level::Trace) {
-            return;
+    #[must_use]
+    pub fn dump_place(&'a self, place: Place<M::PointerTag>) -> PlacePrinter<'a, 'mir, 'tcx, M> {
+        PlacePrinter { ecx: self, place }
+    }
+
+    #[must_use]
```

Comment:
Why `must_use`?

Reply:
because I forgot to use them when I moved from the side-effecting printing functions to the ones returning a printing handle. I can move the attribute to the types if you prefer.

Reply:
yea this is a drive-by fix. It not having side effects means the result should be used

Reply:
Ah I see, makes sense.